### PR TITLE
Help resim cli users get autocomplete

### PR DESCRIPTION
**Pull request by SimonBox**
# Description of change

Adds a convenience command that lets the user generate autocomplete scripts
for various shell types. Also adds docs to help the user get autocomplete set up.

## Asana task link

[Asana task](https://app.asana.com/0/1205228215063249/1205227572053874/f)

## Guide to reproduce test results

1. Follow the instruction in the README.
2. Try a few commands, white periodically hitting tab.

## Checklist

- [x] I have self-reviewed this change.
- [x] I have tested this change.
- [ ] This change is covered by tests that are already landed, or in this PR.

### CODE REVIEW

File: cmd/resim/commands/completion.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
+package commands
+
+import (
+	"os"
+
+	"github.com/resim-ai/api-client/api"
```
Comment (mikebauer) [inline cmd/resim/commands/completion.go:6]: I think this is unused so it fails to build when I try.

File: README.md [hunk 2]
```diff
@@ -41,6 +41,17 @@ To call a particular endpoint, use
 
     RESIM_CLIENT_ID=<client ID> RESIM_CLIENT_SECRET=<client secret> resim create project <flags> 
 
+### Autocomplete
+
+If you would like resim commands to autocomplete you can generate autocomplete scripts using e.g.
+
+    resim complete bash > resim_bash_completion
```
Comment (mikebauer) [inline README.md:48]:
```suggestion
    resim completion bash > resim_bash_completion
```

**Review by mikebauer — CHANGES_REQUESTED** [review]
Couple things don't really work on my end, but seems to work fine with these changes for me with bash.

File: README.md (hunk 2 above)
Comment (optimality) [reply README.md:48]: If you do this you can omit `completion.go`

**Review by optimality — CHANGES_REQUESTED** [review]

**Review by mikebauer — APPROVED** [review]

**Review by optimality — APPROVED** [review]

**Pull request merged**